format/format.go

```diff
@@ -1356,3 +1386,36 @@ func ensureFutureKeywordImport(imps []*ast.Import, kw string) []*ast.Import {
 	imp.Location = defaultLocation(imp)
 	return append(imps, imp)
 }
+
+// ArgErrDetail but for `fmt` checks since compiler has not run yet.
+type ArityFormatErrDetail struct {
+	Have []string `json:"have"`
+	Want []string `json:"want"`
+}
+
+// arityMismatchError but for `fmt` checks since the compiler has not run yet.
+func ArityFormatMismatchError(operands []*ast.Term, operator string, loc *ast.Location, f *types.Function) *ast.Error {
+	want := make([]string, len(f.Args()))
+	for i := range f.Args() {
+		want[i] = types.Sprint(f.Args()[i])
+	}
+
+	have := make([]string, len(operands))
+	for i := 0; i < len(operands); i++ {
+		have[i] = ast.TypeName(operands[i].Value)
+	}
+	err := ast.NewError(ast.TypeErr, loc, "%v: %v", operator, "arity mismatch")
```

Comment:
[nit] it's only strings
```suggestion
	err := ast.NewError(ast.TypeErr, loc, "%s: %s", operator, "arity mismatch")
```